### ISSUE TITLE
Absolute duality gap stopping criterion

### ISSUE THREAD

**Issue by stephane-caron** [Simple-Robotics/proxsuite#168]
Looking at [these preliminary results](https://github.com/stephane-caron/qpsolvers_benchmark/blob/0cf7d62f61667364c8ececa44a3e9b60f057b811/results/maros_meszaros_dense.md), ProxQP appears less good than the other solvers in enforcing the duality gap (which is also affecting its success rate). It could be however that the solver is converging fine, but its internal duality-gap absolute stopping criterion is relaxed compared to the contract :point_down: 

(Sorry if I'm missing something, just checking :blush:)

In both backends there is a test:

```cpp
    if (is_primal_feasible && is_dual_feasible) {
      if (qpsettings.check_duality_gap) {
        if (std::abs(qpresults.info.duality_gap) <=
            qpsettings.eps_abs + qpsettings.eps_rel * rhs_duality_gap) {
```

Both lhs and rhs duality are scaled by ``sqrt_max_dim`` in the backend's ``utils.hpp``:

```cpp
  duality_gap /= sqrt_max_dim; // in order to get an a-dimensional duality gap
  rhs_duality_gap /= sqrt_max_dim;
```

So the relative stopping criterion is the same as in e.g. SCS. But then isn't the absolute stopping criterion relaxed to $r_g \leq \epsilon_{abs} \sqrt{d}$ instead of $r_g \leq \epsilon_{abs}$? (Denoting by $r_g$ the unscaled duality gap.)

I understand there is an argument about making the gap adimensional. However, this makes benchmarking ProxQP against other solvers more difficult, unless there were a way to set ``eps_abs_gap`` and ``eps_rel_gap`` (noooo :see_no_evil:)

**Comment by jcarpent** [comment]
Thanks @stephane-caron for raising this important discussion.
The most reasonable choice will be to add specific stopping accuracy (absolute and relative) for the duality gap.

@Bambade Do you agree?

**Comment by jcarpent** [comment]
Solved via #169.
